aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -294,11 +423,15 @@ module aptos_framework::transaction_validation {
             transaction_fee_amount
         };
 
-        if (amount_to_burn > storage_fee_refunded) {
-            let burn_amount = amount_to_burn - storage_fee_refunded;
+        let refund = storage_fee_refunded;
```

Comment:
assertion on line 407 seems not proper any more

also, line 417 is probably broken

Reply:
why? i thought the only change to the epilogue is to mint back?

Reply:
We should consider the burnt amount when judging if the sender has enough fund to cover the gas?

Reply:
so refund should happen earilier, somewhere before L407?

Reply:
I think you should do the math with the refund accounted for and still either mint or refund once.